src/plugins/teleop/Teleop.cc

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <iostream>
+#ifdef _MSC_VER
+#pragma warning(push, 0)
+#endif
+#include <ignition/msgs/twist.pb.h>
+#ifdef _MSC_VER
+#pragma warning(pop)
+#endif
+
+#include <string>
+
+#include <ignition/gui/Application.hh>
+#include <ignition/gui/MainWindow.hh>
+#include <ignition/plugin/Register.hh>
+
+#include "Teleop.hh"
+
+namespace ignition
+{
+namespace gui
+{
+namespace plugins
+{
+  enum class keyLinear{
+    forward,
+    backward,
+    stop,
+  };
+
+  enum class keyAngular{
+    left,
+    right,
+    stop,
+  };
+
+  class TeleopPrivate
+  {
+    /// \brief Node for communication.
+    public: ignition::transport::Node node;
+
+    /// \brief Topic. Set '/cmd_vel' as default.
+    public: std::string topic = "/cmd_vel";
+
+    /// \brief Publisher.
+    ignition::transport::Node::Publisher cmdVelPub;
+
+    /// \brief Linear velocity.
+    public: float linearVel = 0;
+    /// \brief Angular velocity.
+    public: float angularVel = 0;
```

Comment:
I think we can use `double` type instead

Reply:
Done